src/BasePaintSubscription.sol

```diff
@@ -48,49 +57,50 @@ contract BasePaintSubscription is Ownable {
     error SubAlreadyExists();
     error SubDoesntExist();
     error WrongAmount();
+    error InsufficientSubsidy();
 
-    IBasePaint public basepaint;
-    address public minter;
+    constructor(address _basepaint, address _minter, address _owner) Ownable(_owner) {
+        basepaint = IBasePaint(_basepaint);
+        minter = _minter;
+    }
 
-    struct Subscription {
-        address mintToAddress;
-        uint96 balance;
-        uint8 mintPerDay;
-        uint16 lastMinted;
-        bool active;
+    function addSubsidy() external payable onlyOwner {
+        if (msg.value == 0) revert MustSendETH();
+        subsidyBalance += msg.value;
+        emit SubsidyAdded(msg.value);
     }
 
-    mapping(address => Subscription) public subscriptions;
+    function withdrawSubsidy(uint256 _amount) external onlyOwner {
+        if (_amount > subsidyBalance) revert InsufficientSubsidy();
+        subsidyBalance -= _amount;
+        (bool success,) = payable(msg.sender).call{value: _amount}("");
+        if (!success) revert RefundFailed();
+    }
 
-    constructor(address _basepaint, address _minter, address _owner) Ownable(_owner) {
-        basepaint = IBasePaint(_basepaint);
-        minter = _minter;
+    function getSubsidizedPrice(uint256 _price) public pure returns (uint256) {
+        return _price * (100 - SUBSIDY_PERCENT) / 100;
     }
 
     function subscribe(uint8 _mintPerDay, address _mintToAddress, uint256 _length) external payable {
-        uint256 value = msg.value;
-        if (value == 0) revert MustSendETH();
+        if (msg.value == 0) revert MustSendETH();
         if (_mintPerDay == 0) revert CantBeZero();
-
-        address sender = msg.sender;
-
-        uint256 price = basepaint.openEditionPrice();
-        uint256 total = price * _mintPerDay * _length;
         
-        if (value != total) revert WrongAmount();
-
-        Subscription storage sub = subscriptions[sender];
+        Subscription storage sub = subscriptions[msg.sender];
         if (sub.active) revert SubAlreadyExists();
 
-        address mintTo = _mintToAddress == address(0) ? sender : _mintToAddress;
-
-        sub.mintToAddress = mintTo;
-        sub.balance = uint96(value);
+        uint256 price = basepaint.openEditionPrice();
+        uint256 subsidizedPrice = getSubsidizedPrice(price);
+        uint256 total = subsidizedPrice * _mintPerDay * _length;
+        
+        if (msg.value != total) revert WrongAmount();
```

Comment:
Not a fan of this, just let the user top up whatever amount of ETH they want. Price changes in BasePaint contract means that there could be a situation where the amount of ETH they have left is != whole number of canvases